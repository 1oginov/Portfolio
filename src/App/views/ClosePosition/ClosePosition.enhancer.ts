import { RouteComponentProps as WithRouterProps, withRouter } from 'react-router-dom';
import { compose, withHandlers } from 'recompose';

import withPositionById, { Props as WithPositionByIdProps } from 'Portfolio/enhancers/withPositionById';

import { Props } from './ClosePosition';
import * as R from '../../routes';

export default compose<Props, {}>(
  withRouter,
  withPositionById<WithRouterProps<{ id: string }>>(({ match: { params: { id } } }) => id),
  withHandlers<WithRouterProps & WithPositionByIdProps, {}>({

    handleBackClick: ({ history, position }) => () => {
      if (position) {
        history.push(R.toPosition(position.id));
      }
    },

  }),
);
