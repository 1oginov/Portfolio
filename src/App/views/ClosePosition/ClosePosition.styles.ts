import { withStyles } from '@material-ui/core';

export default withStyles(theme => ({
  back: {
    left: theme.spacing(6 - 3),
    position: 'absolute',
    top: theme.spacing(6 + 3) - (48 - 39) / 2, // Back button height - headline width.
  },
  headline: {
    marginBottom: theme.spacing(3),
    marginLeft: 48 - theme.spacing(3), // Back button width.
    marginRight: 48 - theme.spacing(3), // Back button width.
    marginTop: theme.spacing(3),
    textAlign: 'center',
  },
  root: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    padding: theme.spacing(6),
    position: 'relative',
  },
}));
