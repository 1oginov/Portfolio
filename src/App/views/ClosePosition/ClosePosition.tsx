import { IconButton, Typography } from '@material-ui/core';
import { ArrowBackOutlined } from '@material-ui/icons';
import * as React from 'react';

import ClosePositionForm from 'Portfolio/components/ClosePositionForm';
import { Position } from 'Portfolio/lib';
import Progress from 'Shared/components/Progress';

export interface Props {
  classes: { [key: string]: string };
  handleBackClick: () => void;
  position: Position | null;
  positionLoading: boolean;
}

const ClosePosition: React.FunctionComponent<Props> = ({
  classes, handleBackClick, position, positionLoading,
}: Props) => (
  <div className={classes.root}>

    <IconButton className={classes.back} onClick={handleBackClick}><ArrowBackOutlined /></IconButton>

    <Typography className={classes.headline} variant="h4">Close position</Typography>

    {positionLoading || !position
      ? <Progress />
      : <ClosePositionForm id={position.id} onClose={handleBackClick} />}

  </div>
);

export default ClosePosition;
