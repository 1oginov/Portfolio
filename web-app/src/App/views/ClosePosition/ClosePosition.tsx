import { Typography } from '@material-ui/core';
import * as React from 'react';

import { Position } from 'Layers/Business/Services/PortfolioService/Interfaces/Position';
import { Progress } from 'Layers/Presentation/Components/Progress';
import { ClosePositionForm } from 'Layers/Presentation/Components/ClosePositionForm';

export interface Props {
  classes: { [key: string]: string };
  handleBackClick: () => void;
  position: Position | null;
  positionLoading: boolean;
}

const ClosePosition: React.FunctionComponent<Props> = ({
  classes, handleBackClick, position, positionLoading,
}: Props) => (
  <div className={classes.root}>

    <Typography className={classes.headline} variant="h4">Close position</Typography>

    {positionLoading || !position
      ? <Progress />
      : (
        <ClosePositionForm
          onCancelButtonClick={handleBackClick}
          onClose={handleBackClick}
          position={position}
        />
      )}

  </div>
);

export default ClosePosition;
