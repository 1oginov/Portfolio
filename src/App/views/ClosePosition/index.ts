import ClosePosition from './ClosePosition';
import enhance from './ClosePosition.enhancer';

export default enhance(ClosePosition);
