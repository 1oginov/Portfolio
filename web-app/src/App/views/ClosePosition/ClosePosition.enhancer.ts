import { RouteComponentProps as WithRouterProps, withRouter } from 'react-router-dom';
import { compose, withHandlers } from 'recompose';

import {
  PositionByIdConnector, PositionByIdConnectorProps,
} from 'Layers/Adapter/Connectors/PositionByIdConnector/PositionByIdConnector';
import {
  PositionsFirestoreConnector,
} from 'Layers/Adapter/FirestoreConnectors/PositionsFirestoreConnector/PositionsFirestoreConnector';

import { Props } from './ClosePosition';
import * as R from '../../routes';

interface WithHandlersProps {
  handleBackClick: () => void;
}

export default compose<Props, Record<string, never>>(
  withRouter,
  PositionsFirestoreConnector,
  PositionByIdConnector<WithRouterProps<{ id: string }>>(({ match: { params: { id } } }) => id),
  withHandlers<WithRouterProps & PositionByIdConnectorProps, WithHandlersProps>({

    handleBackClick: ({ history, position }) => () => {
      if (position) {
        history.push(R.toPosition(position.id));
      }
    },

  }),
);
