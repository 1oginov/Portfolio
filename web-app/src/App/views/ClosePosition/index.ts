import ClosePosition from './ClosePosition';
import enhance from './ClosePosition.enhancer';
import style from './ClosePosition.styles';

export default style(enhance(ClosePosition));
