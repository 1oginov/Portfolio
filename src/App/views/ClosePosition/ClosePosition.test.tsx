import { shallow } from 'enzyme';
import * as React from 'react';

import ClosePosition from './ClosePosition';

it('matches snapshot', () => {
  const wrapper = shallow(
    <ClosePosition
      handleBackClick={() => undefined}
      position={{
        amount: 1,
        closeCommission: null,
        closeDate: null,
        closePrice: null,
        id: 'id',
        openCommission: 1,
        openDate: '2019-07-01',
        openPrice: 100,
        symbol: 'AAPL',
      }}
      positionLoading={false}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when loading', () => {
  const wrapper = shallow(
    <ClosePosition
      handleBackClick={() => undefined}
      position={null}
      positionLoading
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
