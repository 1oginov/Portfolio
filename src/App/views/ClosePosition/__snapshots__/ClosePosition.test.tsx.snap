// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="root"
>
  <WithStyles(ForwardRef(Typography))
    className="headline"
    variant="h4"
  >
    Close position
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(Connect(withStateHandlers(withHandlers(ClosePositionForm))))
    backButton={
      <ForwardRef(WithStyles)
        color="primary"
        onClick={[Function]}
      >
        Cancel
      </ForwardRef(WithStyles)>
    }
    id="id"
    onClose={[Function]}
  />
</div>
`;

exports[`matches snapshot when loading 1`] = `
<div
  className="root"
>
  <WithStyles(ForwardRef(Typography))
    className="headline"
    variant="h4"
  >
    Close position
  </WithStyles(ForwardRef(Typography))>
  <Progress />
</div>
`;
